### PR TITLE
Add 30 seconds timeout for `docker stop`

### PR DESCRIPTION
**Pull request by dims**
Seeing some odd behavior where docker is refusing to come down and the prow job just sits there (waiting for it to die) even after the script has finished running. Hoping if we can do a timeout, that will do the trick?

Log snippet from [here](https://storage.googleapis.com/kubernetes-jenkins/logs/ci-kubernetes-local-e2e/1741211742429515776/build-log.txt)
```
2023/12/30 23:39:38 process.go:155: Step 'bash -c . hack/lib/version.sh && KUBE_ROOT=. kube::version::get_version_vars && echo "${KUBE_GIT_VERSION-}"' finished in 6.087183ms
+ EXIT_VALUE=0
+ set +o xtrace
Cleaning up after docker in docker.
================================================================================
Cleaning up after docker
Stopping Docker: dockerProgram process in pidfile '/var/run/docker-ssd.pid', 1 process(es), refused to die.
================================================================================
Done cleaning up after docker in docker.
{"component":"entrypoint","file":"k8s.io/test-infra/prow/entrypoint/run.go:169","func":"k8s.io/test-infra/prow/entrypoint.Options.ExecuteProcess","level":"error","msg":"Process did not finish before 4h0m0s timeout","severity":"error","time":"2023-12-31T01:36:48Z"}
{"component":"entrypoint","error":"os: process already finished","file":"k8s.io/test-infra/prow/entrypoint/run.go:255","func":"k8s.io/test-infra/prow/entrypoint.gracefullyTerminate","level":"error","msg":"Could not interrupt process after timeout","severity":"error","time":"2023-12-31T01:36:48Z"}
{"component":"entrypoint","file":"k8s.io/test-infra/prow/entrypoint/run.go:267","func":"k8s.io/test-infra/prow/entrypoint.gracefullyTerminate","level":"error","msg":"Process did not exit before 15m0s grace period","severity":"error","time":"2023-12-31T01:51:48Z"}
{"component":"entrypoint","error":"os: process already finished","file":"k8s.io/test-infra/prow/entrypoint/run.go:269","func":"k8s.io/test-infra/prow/entrypoint.gracefullyTerminate","level":"error","msg":"Could not kill process after grace period","severity":"error","time":"2023-12-31T01:51:48Z"}
```

### CODE REVIEW

File: images/bootstrap/runner.sh [hunk 1]
```diff
@@ -29,7 +29,8 @@ cleanup_dind() {
     if [[ "${DOCKER_IN_DOCKER_ENABLED:-false}" == "true" ]]; then
         echo "Cleaning up after docker"
         docker ps -aq | xargs -r docker rm -f || true
-        service docker stop || true
+        echo "Waiting for docker to stop for 30 seconds"
+        timeout 30 service docker stop || true
```
Comment (BenTheElder) [inline images/bootstrap/runner.sh:33]: `service docker stop` is already exiting though?

Comment (BenTheElder) [reply images/bootstrap/runner.sh:33]: `timeout` would terminate `service docker stop` if it hangs, but what we're actually seeing in the logs is some internal failure attempting to kill dockerd?

Comment (dims) [reply images/bootstrap/runner.sh:33]: it was exiting, but i think it took forever (looking at timestamps around it).

I am happy to try something else as well if this is not kosher.

Comment (BenTheElder) [reply images/bootstrap/runner.sh:33]: Ahh! Got it.

Comment (BenTheElder) [reply images/bootstrap/runner.sh:33]: We should also do this in images/krte/wrapper.sh

/lgtm
/approve
/hold

Comment (dims) [reply images/bootstrap/runner.sh:33]: good call @BenTheElder, updated wrapper.sh now.

**Comment by aojea** [comment]
probably related https://github.com/moby/moby/issues/43094 ?

/lgtm

**Review by BenTheElder — COMMENTED** [review]
/lgtm
/approve

**Comment by dims** [comment]
/hold cancel

thanks!

**Pull request merged**